workflow-ui/internal-testing-android/src/main/java/com/squareup/workflow1/ui/internal/test/DetectLeaksAfterTestSuccess.kt

```diff
@@ -37,3 +38,15 @@ public class DetectLeaksAfterTestSuccess(
     }
   }
 }
+
+/**
+ * Invokes [LeakAssertions.assertNoLeaks] before performing the tear-down logic of the receiver
+ * [TestRule], then again *after* the other rule's logic.
+ *
+ * https://github.com/square/workflow-kotlin/issues/657
+ */
+public fun TestRule.wrapInLeakCanary(): RuleChain = requireNotNull(
```

Comment:
The actual rule is just above the fold, so GitHub won't let me comment on the line I'm interested in...

https://github.com/square/workflow-kotlin/blob/72ba8c7bedbd0118164db82b49bc849aa119e90c/workflow-ui/internal-testing-android/src/main/java/com/squareup/workflow1/ui/internal/test/DetectLeaksAfterTestSuccess.kt#L29-L36

If we're executing this twice per test, do we want to skip the call to `clearWatchedObjects()` the first time?

Reply:
I have no idea. @py?

Reply:
No, should be fine. But maybe use clearAfter timestamp instead

Reply:
```
 try { 
   base.evaluate()
   val heapDumpUptimeMillis = SystemClock.uptimeMillis()
   LeakAssertions.assertNoLeaks(tag) 
 } finally { 
   // Otherwise upstream test failures will be reported as leaks. 
   // https://github.com/square/leakcanary/issues/2297 
   AppWatcher.objectWatcher.clearObjectsWatchedBefore(heapDumpUptimeMillis) 
 } 
 ```

Reply:
I made this change -- thanks!